### DISCUSSION TITLE
Hide users list from sidebar when logged in as a custom role user

### DISCUSSION THREAD

**Discussion by pkokorev** [directus/directus#3657] [Feature Requests]
I was trying to figure out the following scenario using NodeJS version of the tool.

1. Create a custom role (for customers),
2. Allow customers to log in to the app,
3. Remove / restrict users list in the sidebar so that customers don't see each others info

If I setup the system roles by default, the "Directus Users" permission will be activated for read-only:
![image](https://user-images.githubusercontent.com/5184701/104284998-3e895980-54b3-11eb-9e2d-52d84468a0e7.png)

In this case, after logging in with the custom role user, I can see both this user and the second admin user (that I want to avoid for customers):
![image](https://user-images.githubusercontent.com/5184701/104285074-5cef5500-54b3-11eb-89ea-df2bbe741b38.png)

If I deactivate the read-only permission on "Directus Users" for this role, the app login is no longer possible:
![image](https://user-images.githubusercontent.com/5184701/104285210-8dcf8a00-54b3-11eb-83a4-b0081efd3bc2.png)

Finally, by debugging the network calls, I realized that the login actually happens (response 200) even though the error message on the screen says "no access", so I used the `$CURRENT_USER` variable to restrict the list of users by adding an item permission in the role setup.
```
{
    "id": {
        "_eq": "$CURRENT_USER"
    }
}
```
After doing that, I was able to login to the app using custom role user and the list only shows my current user as expected.

Hope someone finds it helpful.

I still wonder whether it's possible to completely hide the users button from the sidebar with settings, think it would be a useful feature.

**Comment by nelolka** [comment]
Also you can create your own module navigation and add only needed links.
![image](https://user-images.githubusercontent.com/24505418/104385469-f5c9b300-553b-11eb-9a0f-d22d2a71487c.png)

**Reply by david-zacharias** [reply]
Is this still possible in recent versions (I checked with 9.0.0-rc.98) and can only configure groups of collections on role level. I would love to adapt the module list based on roles.

**Reply by benhaynes** [reply]
We'll likely be adding it back in for Roles, but needed to go top-down for now.

**Reply by david-zacharias** [reply]
Since I know you can not tell when this will happen, does a related issue exist, which I can subscribe to get updated?

**Reply by benhaynes** [reply]
No issue, since it's not technically a bug, but we have this (and maybe another) Discussion that should get updated. Also, if you or anyone else desperately needs this back in sooner than we have it planned... it can always be sponsored.

**Reply by sean-nicholas** [reply]
I found this blogpost from @martijnboland that adds this capability with hooks: https://blogs.taiga.nl/martijn/2022/04/29/use-a-hook-to-hide-modules-for-non-admin-roles-in-directus/

**Comment by krazyjakee** [comment]
This is great but sadly you lose the ability to see who created which items in a collection.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
